hw/arm/guest-socket.c

```diff
@@ -49,44 +117,59 @@ int32_t qc_handle_socket(CPUState *cpu, int32_t domain, int32_t type,
         guest_svcs_errno = ENOTSOCK;
     } else if ((guest_svcs_fds[retval] = socket(domain, type, protocol)) < 0) {
         retval = -1;
-        guest_svcs_errno = errno;
+        guest_svcs_errno = convert_error(errno);
     }
 
     return retval;
 }
 
-int32_t qc_handle_accept(CPUState *cpu, int32_t sckt, struct sockaddr *g_addr,
-                         socklen_t *g_addrlen)
+int32_t qc_handle_accept(CPUState *cpu, int32_t sckt, S_SOCKADDR *g_addr,
+                         SOCKLEN_T *g_addrlen)
 {
     struct sockaddr_in addr;
-    socklen_t addrlen;
+    SOCKLEN_T addrlen;
 
+    addrlen = sizeof(addr);
     VERIFY_FD(sckt);
 
     int retval = find_free_socket();
 
     // TODO: timeout
     if (retval < 0) {
         guest_svcs_errno = ENOTSOCK;
-    } else if ((guest_svcs_fds[retval] = accept(guest_svcs_fds[sckt],
+    } else if ((guest_svcs_fds[retval] =
+#ifdef NETWORK_HACK
+                                         accept4(guest_svcs_fds[sckt],
+                                         (struct sockaddr *) &addr,
+                                         &addrlen, SOCK_NONBLOCK)) < 0)
```

Comment:
Maybe use regular `accept` (avoiding the `#ifdef`) and simply set the flag based on what was set on the listening socket?

Reply:
Will suggest changes in another comment.